request_handlers_test.cc

```diff
@@ -1,100 +1,110 @@
+#include "config_parser.h"
 #include "gtest/gtest.h"
+#include "mime_types.h"
 #include "request_handlers.h"
+#include "request_router.h"
+#include "server_config.h"
+
+#include <cstddef>
+#include <fstream>
+#include <sstream>
 
 class RequestHandlersTest : public ::testing::Test {
 protected:
-  const size_t header_size = 45;
+
+  const size_t ok_header_size = 45;
+
   bool HandleRequest(const char* request_buffer,
                      const size_t& request_buffer_size,
                      char* &response_buffer,
                      size_t& response_buffer_size) {
 
-    EchoRequestHandler echo_request_handler;
-    echo_request_handler.handleRequest(request_buffer,
-                                       request_buffer_size,
-                                       response_buffer,
-                                       response_buffer_size);
-    if (response_buffer == nullptr) {
-      return false;
-    }
-    else {
-      return true;
-    }
+    // Read config file
+    NginxConfig config;
+    NginxConfigParser config_parser;
+    config_parser.Parse("simple_config", &config);
+    ServerConfig server_config(config);
+
+    RequestRouter router;
+    return router.routeRequest(server_config,
+                               request_buffer,
+                               request_buffer_size,
+                               response_buffer,
+                               response_buffer_size);
   }
 };
 
+// TODO tests:
+// invalid route
+// invalid file
+
 TEST_F(RequestHandlersTest, EmptyStringTest) {
-  char request_buffer[] = "";
+  const char request_buffer[] = "";
   const size_t request_buffer_size = 0;
 
   // The request_buffer is normally filled by a socket-read
   // The response_buffer is allocated by the request-handler
   char* response_buffer = nullptr;
   size_t response_buffer_size = 0;
 
-  const char expected_response_buffer[] =
-    "HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\n\r\n";
-  const size_t expected_response_buffer_size = header_size + request_buffer_size;
+  const char expected_response_buffer[] = "";
+  const size_t expected_response_buffer_size = 0;
 
-  ASSERT_TRUE(HandleRequest(request_buffer,
-                            request_buffer_size,
-                            response_buffer,
-                            response_buffer_size));
+  ASSERT_FALSE(HandleRequest(request_buffer,
+                             request_buffer_size,
+                             response_buffer,
+                             response_buffer_size));
 
   EXPECT_EQ(response_buffer_size, expected_response_buffer_size)
     << "Response size should only be header size";
```

Comment:
Can we update these test-failure log messages to reflect the particular test failed? Otherwise, we won't be able to distinguish the reasons for different test failures.